cmd/libsnap-confine-private/test-utils.c

```diff
@@ -71,3 +72,33 @@ void rm_rf_tmp(const char *dir)
 	g_free(argv[3]);
 	g_free(argv);
 }
+
+void
+__attribute__((sentinel)) test_argc_argv(int *argcp, char ***argvp, ...)
+{
+	int argc = 0;
+	char **argv = NULL;
+	g_test_queue_free(argv);
+
+	va_list ap;
+	va_start(ap, argvp);
+	const char *arg;
+	do {
+		arg = va_arg(ap, const char *);
```

Comment:
You can also go over the arguments once and count them, malloc once, go over the arguments again and assign them.

Reply:
Yeah, since we're moving the code around, why not fix it a bit too.